e2e-test/src/tests/upload.test.js

```diff
@@ -0,0 +1,44 @@
+import { vBtn } from 'jest-puppeteer-vuetify';
+import { CLIENT_URL } from '../util';
+
+
+describe('upload dataset', () => {
+  it('upload one dataset and verify it', async () => {
+    await Promise.all([
+      page.goto(CLIENT_URL),
+      page.waitForNavigation({ waitUntil: 'networkidle0' }),
+    ]);
+    await expect(page).toClickXPath(vBtn('Your Datasets'));
+    const fileInputElementHandle = await page.waitForXPath("//input[@type='file']");
+    await fileInputElementHandle.uploadFile('../e2e-test/data/Cancer chemo muscle (NMR).csv');
+    const fileName = await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]");
+    expect(fileName).not.toBeNull();
```

Comment:
I'm not sure about this approach. If there is the file that matches the name, then the return result won't be null.
I have tried the toHaveLength approach but it didn't work as I expected.
![Selection_069](https://user-images.githubusercontent.com/59888938/87822252-23f56100-c83f-11ea-8a62-8329cdc95868.png)

Reply:
[`page.waitForXPath`](https://devdocs.io/puppeteer/index#pagewaitforxpathxpath-options) returns the first element identified by an XPath, or throws an error if no matching element could be found within the timeout.
An error is thrown if it can't find the element, so it's enough to just call `waitForXPath`. The null check is technically dead code, since `waitForXPath` will throw an error instead of returnning null.
It will also make the assert more readable since you don't need the extra line, which is a nice bonus.